### PR TITLE
Try: Ensure compatibility with WordPress 5.0

### PR DESCRIPTION
**Pull request by mcsf**
This lets the plugin run alongside WP trunk and override core packages.

I recommend seeing the [diff without whitespace changes](https://github.com/WordPress/gutenberg/pull/10754/files?utf8=✓&diff=unified&w=1).

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation.

### CODE REVIEW

File: lib/blocks.php [hunk 1]
```diff
@@ -40,8 +42,10 @@ function register_block_type( $name, $args = array() ) {
  *                                   complete WP_Block_Type instance.
  * @return WP_Block_Type|false The unregistered block type on success, or false on failure.
  */
-function unregister_block_type( $name ) {
-	return WP_Block_Type_Registry::get_instance()->unregister( $name );
+if ( ! function_exists( 'register_block_type' ) ) {
```
Comment (ZebulanStanphill) [inline lib/blocks.php]: Why not `if ( ! function_exists( 'unregister_block_type' ) )`?

Comment (mcsf) [reply lib/blocks.php]: That was a typo, thanks for noticing.

File: lib/blocks.php [hunk 2]
```diff
@@ -9,129 +9,139 @@
...
-				'innerBlocks' => array(),
-				'innerHTML'   => $content,
-			),
-		);
+	function unregister_block_type( $name ) {
+		return WP_Block_Type_Registry::get_instance()->unregister( $name );
 	}
+}
 
+if ( ! function_exists( 'gutenberg_parse_blocks' ) ) {
```
Comment (pento) [inline lib/blocks.php]: I've renamed the function to `parse_blocks()` in Core, it'd be good to mark `gutenberg_parse_blocks()` as `_doing_it_wrong()`.

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Comment by gziolo** [comment]
Just out of curiosity, what about all `gutenberg_` prefixes? When do we plan to remove them from core?

File: lib/blocks.php (hunk 2 above)
Comment (mcsf) [reply lib/blocks.php]: Thanks for the tip. Would `_deprecated_function` be more apt?

Btw, I'm noticing that the Codex states that both are private and not intended for plugin use. ¯\_(ツ)_/¯

**Comment by mcsf** [comment]
> Just out of curiosity, what about all gutenberg_ prefixes? When do we plan to remove them from core?

@gziolo: the prefixes are generally being dropped as pieces are merged in core, and in other cases `gutenberg_` becomes `wp_`. That said, I've spotted at least one instance of slippage (e.g. `gutenberg_render_block_core_latest_comments`) which should be addressed.

**Pull request merged**